sdk/core/azure-core/src/main/java/com/azure/core/implementation/serializer/HttpResponseBodyDecoder.java

```diff
@@ -48,6 +49,7 @@
     static Mono<Object> decode(HttpResponse httpResponse, SerializerAdapter serializer,
                                HttpResponseDecodeData decodeData) {
         ensureRequestSet(httpResponse);
+        final ClientLogger logger = new ClientLogger(HttpResponseBodyDecoder.class);
```

Comment:
Creating a new instance every time could get expensive (if there was complex initialisation logic). But sadly, this is a static method/class.